### PR TITLE
PAM config module

### PR DESCRIPTION
**Pull request by juliedavila**
This a module for managing PAM configs.

### CODE REVIEW

File: system/pam.py [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+            if set( re.split(r'[\s\t\n]*(?![^\[\]]*\])', config_lines[ref_location + 1]) ) == set(entry_args):
+                changed = False
+            else:
+                config_lines.insert(ref_location + 1, pam_entry + '\n')
+                changed = True
+        except IndexError:
+            config_lines.append(pam_entry + '\n')
+            changed = True
+
+    with open(config_path, 'w') as conf:
```
Comment (bcoca) [inline system/pam.py]: won't work with 2.4

**Comment by gregdek** [comment]
Why does this belong in core and not extras?

**Comment by juliedavila** [comment]
@gregdek my thought process has to do with the declared intent to make extras a separately installed 'thing' in the future. Assuming that is still the case, it would be prudent to have this in core, imo, as a big user base of this will be public sector, especially those that leverage the Lockdown project. I can foresee occasional public sector types being limited to only core modules.

**Comment by juliedavila** [comment]
If that's a problem, of course I can resubmit to extras

**Comment by juliedavila** [comment]
Found a couple of bugs, will update when ready

**Comment by juliedavila** [comment]
Updated

**Comment by juliedavila** [comment]
will resubmit later

**Pull request closed without merging**